File: ModelReservaSalas/ModelReservaSalas/Modelo/Pregunta.cs
using System.ComponentModel.DataAnnotations;

namespace ModelReservaSalas.modelo
{
    public abstract class Pregunta
    {
        public static readonly int _DESARROLLO = 1;
        public static readonly int _CHECKBOX = 2;

        public Pregunta(string enunciado, bool required)
        {
            Enunciado = enunciado;
            Required = required;
        }
        public Pregunta() { }

        abstract public bool esCheckbox();

        public int PreguntaID { get; set; }

        [Required]
        public string Enunciado { get; set; }

        public bool Required { get; set; }

        public int TipoPregunta { get; set; }


    }
}

File: ClienteGestionReservaSalas/src/app/shared/model/Pregunta.ts
import { Opcion } from './Opcion';

export class Pregunta {

    public static readonly preguntaDesarrollo = 1;
    public static readonly preguntaCheckbox = 2;

    preguntaID : number;
    enunciado : string;
    tipoPregunta : number;
    required: boolean;
    opciones?: Opcion[];
}
